system/Database/BaseBuilder.php

```diff
@@ -1260,6 +1260,7 @@ public function orderBy(string $orderBy, string $direction = '', ?bool $escape =
         if ($direction === 'RANDOM') {
             $direction = '';
             $orderBy   = ctype_digit($orderBy) ? sprintf($this->randomKeyword[1], $orderBy) : $this->randomKeyword[0];
+            $escape    = false;
```

Comment:
@kenjis 

Should we split the code here?

The intention of this fix is because escaping is not necessary for RANDOM.

In ORACLE, random numbers are `DBMS_RANDOM.RANDOM`. That is, it refers to the `RANDOM` column of the `DBMS_RANDOM` table.
In MySQL, it is `RAND()`, it is function.

The third argument `$escape` of orderBy is null if it is omitted.
It depends on the `protectIdentifiers` property of Connection to toggle whether escaping is performed or not.
Also, the escaping process is to add the prefix to the table name.

In other words, in the case of ORACLE, it will add the prefix to  `DBMS_RANDOM.RANDOM` and execute a query for a non-existent table, which will fail.

It is not a good idea for users to specify the value of the third argument escape with the driver in mind.

Therefore, I am making this fix because I thought it would be better to not escape for RANDOM and add the escaped column name to the `randomKeyword` property of each driver.

This is not limited to Oracle, but in the Database supported by CodeIgniter4 at this time, RANDOM is all function calls and cannot be reproduced.

Reply:
The more you split this PR, the more likely it is to be merged.
I'm not sure, how long it will take it to be merged.
Because this PR is too big for anyone to fully review.

Reply:
@kenjis 

I understand.
Thank you.
I will do my best to remove all diffs from this PR except `system/Database/OCI8`.💪

Reply:
splited #5179